website/content/docs/intro/use-cases/dynamic-config-vault-dynamic-secrets.mdx

```diff
@@ -0,0 +1,269 @@
+}
+
+variable "image_tag" {
+  decription = "Tag of the Docker image for the app built by Waypoint. This will be used with the name to push an image to a Docker repository."
+  type = string
+}
+
+variable "registry_username" {
+  description = "Username of user with push access to your Docker registry."
+  type      = string
+  sensitive = true
+}
+
+variable "registry_password" {
+  description = "Password for the configured user with push access to your Docker registry."
+  type      = string
+  sensitive = true
+}
+```
+
+In the configuration above, we’ve informed Waypoint that we want the Waypoint Entrypoint,
+which will wrap the entrypoint to the Docker container that the Pack plugin is
+building, to get a secret from Vault at the path `database/creds/my-role`. The
+values for the keys `username` and `password` at that secret path will be configured
+as environment variables, `USERNAME` and `PASSWORD`, in the deployment of the app. We
+can confirm that the secret was successfully templated after running `waypoint up`,
+by checking the environment variables using `waypoint exec`, and checking the application
+logs using `waypoint logs`:
+
+```shell
+// To run this operation, I exported my input variables to my environment, prefixed with “WP_VAR_”
+// The output below is abbreviated for brevity
+$ waypoint up
+✓ Running build v2
+✓ Building Buildpack with kaniko...
+✓ Repository is available and ready: devopspaladin/hello-database:latest
+✓ Executing kaniko...
+…
+✓ Image pushed to 'devopspaladin/hello-database:latest'
+✓ Running push build v1
+✓ Running deploy v2
+✓ Job registration successful
+✓ Allocation "4a0788ae-a6c0-bf55-4a84-46a4f0ff4a81" created: node "b9a637ea-6098-4b22-6f52-9d5b94581a75", group
+"dynamic-go-01gj1s1sqj31r72qn38t09xj7p"
+✓ Evaluation status changed: "pending" -> "complete"
+✓ Evaluation "d1c03461-fddd-2e68-bb7b-d6e7e718f121" finished with status "complete"
+✓ Deployment successfully rolled out!
+✓ Job "dynamic-go-01gj1s1sqj31r72qn38t09xj7p" is reporting ready!
+✓ Finished building report for Nomad platform
+
+// The values for USERNAME and PASSWORD below are dynamic credentials generated by Vault!
+$ waypoint exec env
+…
+USERNAME=v-token-readonly-wOdEtyXQ5MktQ5SwTnXo-1668702202
+PASSWORD=e-4QVqEpUhGfmpTncKTy
+…
+
+$ waypoint logs
+...
+2022-11-17T16:41:36.491Z D60TJX: 2022/11/17 16:41:36 Successfully connected to the database! :)
+```
+
+With the application successfully deployed and configured with dynamic credentials,
+it can securely connect to the database, and we never needed to provide the app
+with an actual secret value. The Waypoint Entrypoint, built-in to the application image,
+can now dynamically source this Vault configuration wherever the app is deployed.
+If I deploy to Docker on my laptop for some local development testing, or my EKS
+cluster in AWS, as long as the application container can access the Vault cluster,
+the secrets necessary to run the application will be provided by Waypoint.
```

Comment:
Maybe wrap this up with further reading, learn guide links, links to other docs for more info, etc?